src/python/pants/backend/jvm/tasks/nailgun_task.py

```diff
@@ -95,10 +95,6 @@ def runjava(self, classpath, main, jvm_options=None, args=None, workunit_name=No
     """
     executor = self.create_java_executor(dist=dist, force_subprocess=force_subprocess)
```

Comment:
So, using a synthetic jar means that zinc wont be able to parse the classpath for scala deps, unless we were to parse the synthetic jar manifest in zinc to look for the right paths, but then we would want to parse all jars on the path, because there is nothing special about the synthetic jar per se. I'm a bit dubious that we will ever hit the 128kB arg max limit with the zinc classpath, even with a few new scala jars added, but I'm not completely sure of the backstory of this feature.